openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -74,12 +71,7 @@ def block_metadata(request, usage_key_str):
     * "include": a comma-separated list of keys to include.
       Valid keys are "index_dictionary" and "student_view_data".
     """
-    try:
-        usage_key = UsageKey.from_string(usage_key_str)
-    except InvalidKeyError as e:
-        raise NotFound(invalid_not_found_fmt.format(usage_key=usage_key_str)) from e
-
-    block = load_block(usage_key, request.user)
+    block = load_block(usage_key, request.user, version=parse_version_request(version))
```

Comment:
[Nit, optional]: Per your earlier convention, the `version` param passed into this view could be renamed `version_str` to clarify that it hasn't been parsed yet, or you could create another URL param converter to do the translation before it invokes the view.

Reply:
Sure - I pushed a commit with the URL param converter. I don't know if it's any cleaner than just using `version_str`, but I agree that doing one or the other is an improvement. Thx!